pkg/kapp/app/recorded_app.go

```diff
@@ -522,7 +531,12 @@ func (a *RecordedApp) LastChange() (Change, error) {
 }
 
 func (a *RecordedApp) BeginChange(meta ChangeMeta) (Change, error) {
-	change, err := NewRecordedAppChanges(a.nsName, a.name, a.coreClient).Begin(meta)
+	appMeta, err := a.meta()
+	if err != nil {
+		return nil, err
+	}
```

Comment:
for these additional meta calls, im assuming we arent making additional API calls?

Reply:
For most cases we don't make an additional API call. 
In some cases there would be an additional API call (for example `kapp app-change` commands).

Reply:
I initially added `setMeta()` in a couple of places to make sure that we don't make an unnecessary call, but I reverted it because it would probably make sense to add that in additional PRs. (This won't effect `kapp app-change` commands though)